### PR TITLE
Resolve conflicts between logical and display names in connectors

### PR DESCRIPTION
**Pull request by LucGenetier**
"ServiceDeskPlus Cloud" connector generate failures with a combination of props like this
- logical name = "held_by", display name = "technician"
- logical name = "technician", display name = "technician"

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by MikeStall** [comment]
Can you add a PR description? 
What policy are you using to resolve?  What does Power Apps do?

**Comment by LucGenetier** [comment]
Done.
Logical names have priority and won't be renamed. Only display names could.
Power Apps doesn't support display names properly

---
In reply to: [2090624811](https://github.com/microsoft/Power-Fx/pull/2362#issuecomment-2090624811) [](http://example.com/codeflow?ancestors=2090624811)

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs [hunk 1]
```diff
@@ -623,20 +626,45 @@ internal static ConnectorType GetConnectorType(this OpenApiParameter openApiPara
...
+        internal static RecordType ToRecordType(this List<(string logicalName, string displayName, FormulaType type)> fields)
         {
-            int i = 0;
-            string displayName2 = displayName;
+            if (fields == null)
+            {
+                return null;
+            }
+            
+            HashSet<string> names = new HashSet<string>(fields.Select(f => f.logicalName), StringComparer.InvariantCulture);
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs:636]: This looks like very similar logic to DisplayNameUtility.MakeUnique?

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs (hunk 1 above)
Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs:636]: Yes
The names that we generate are different though

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
Changes already merged, closing PR

**Pull request closed without merging**